### PR TITLE
Change constraints to strings

### PR DESCRIPTION
**Pull request by thecharlieblake**

### CODE REVIEW

File: unit_scaling/_modules.py [hunk 1]
```diff
@@ -27,7 +26,7 @@
 class GELU(nn.GELU):
     def __init__(
         self,
-        constraint: Optional[BinaryConstraint] = gmean,
+        constraint: Optional[str] = "gmean",
```
Comment (DouglasOrr) [inline unit_scaling/_modules.py:29]: Slightly sad to lose the `BinaryConstraint` type info here, but oh well!

File: unit_scaling/docs.py [hunk 2]
```diff
@@ -180,24 +180,29 @@ def f(cls: T) -> T:
...
-    "constraint (Optional[BinaryConstraint], optional): Function which"
-    " takes `output_scale` and `grad_input_scale` and returns a single"
-    " 'constrained' scale (usually necessary for valid gradients). If `None` is"
-    " provided, no constraint will be applied. Defaults to `gmean`."
+    "constraint (Optional[str], optional): The name of the constraint function to be"
+    " applied to the outputs & input gradient. In this case, the constraint name must"
+    " be one of:"
+    " [None, 'gmean', 'hmean', 'amean', 'to_output_scale', 'to_grad_input_scale']"
+    " (see `unit_scaling.constraints` for details on these constraint functions)."
+    " Defaults to `gmean`."
```
Comment (DouglasOrr) [inline unit_scaling/docs.py:188]: :heart: for the docs, without the type hint, these are even more important.

File: unit_scaling/functional.py [hunk 3]
```diff
@@ -131,12 +132,9 @@ def matmul(
     left_grad_scale = right_size**-0.5
     right_grad_scale = left_size**-0.5
 
-    if constraint:
-        scale = constraint(output_scale, left_grad_scale, right_grad_scale)
-        if isinstance(scale, Sequence):
-            output_scale, left_grad_scale, right_grad_scale = scale  # type: ignore
```
Comment (DouglasOrr) [inline unit_scaling/functional.py:137]: When did we previously return a sequence? Can't see any in `constraints`.

**Review by DouglasOrr — APPROVED** [review]
LGTM; I minded the change less as I read more—all good!

File: unit_scaling/functional.py (hunk 3 above)
Comment (thecharlieblake) [reply unit_scaling/functional.py:137]: Yeah I think I had a custom constraint that did this (back when that was allowed). Might leave in for now just to give the option of adding constraints later that do?

**Pull request merged**